staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -251,256 +272,132 @@ func TestApplyDoesNotStripLabels(t *testing.T) {
+			obj: getObjectBytes("pod.yaml"),
 		},
-
-	}`
-	obj1 := &unstructured.Unstructured{Object: map[string]interface{}{}}
-	if err := yaml.Unmarshal([]byte(y1), &obj1.Object); err != nil {
-		b.Fatalf("Failed to parse yaml object: %v", err)
-	}
-	y2 := `{
-		"apiVersion": "apps/v1",
-		"kind": "Deployment",
-		"metadata": {
-			"name": "b",
-			"namespace": "b",
-			"creationTimestamp": "2016-05-19T09:59:00Z",
+		{
+			gvk: schema.FromAPIVersionAndKind("v1", "Node"),
+			obj: getObjectBytes("node.yaml"),
 		},
-                "map": {
-                        "fieldA": 1,
-                        "fieldB": 1,
-                        "fieldC": 1,
-                        "fieldD": 1,
-                        "fieldE": 1,
-                        "fieldF": 1,
-                        "fieldG": 1,
-                        "fieldH": 1,
-                        "fieldI": 1,
-                        "fieldJ": 1,
-                        "fieldK": 1,
-                        "fieldL": 1,
-                        "fieldM": 1,
-                        "fieldN": {
-	                        "fieldN": {
-					"fieldN": {
-						"fieldN": {
-				                        "fieldN": {
-								"value": false
-							},
-						},
-					},
-				},
-			},
+		{
+			gvk: schema.FromAPIVersionAndKind("v1", "Endpoints"),
+			obj: getObjectBytes("endpoints.yaml"),
 		},
-
-	}`
-	obj2 := &unstructured.Unstructured{Object: map[string]interface{}{}}
-	if err := yaml.Unmarshal([]byte(y2), &obj2.Object); err != nil {
-		b.Fatalf("Failed to parse yaml object: %v", err)
 	}
-	y3 := `{
-		"apiVersion": "apps/v1",
-		"kind": "Deployment",
-		"metadata": {
-			"name": "b",
-			"namespace": "b",
-			"creationTimestamp": "2016-05-19T09:59:00Z",
-		},
-                "map": {
-                        "fieldA": 1,
-                        "fieldB": 1,
-                        "fieldC": 1,
-                        "fieldD": 1,
-                        "fieldE": 1,
-                        "fieldF": 1,
-                        "fieldG": 1,
-                        "fieldH": 1,
-                        "fieldI": 1,
-                        "fieldJ": 1,
-                        "fieldK": 1,
-                        "fieldL": 1,
-                        "fieldM": 1,
-                        "fieldN": {
-	                        "fieldN": {
-					"fieldN": {
-						"fieldN": {
-				                        "fieldN": {
-								"value": true
-							},
-						},
-					},
-				},
-			},
-                        "fieldO": 1,
-                        "fieldP": 1,
-                        "fieldQ": 1,
-                        "fieldR": 1,
-                        "fieldS": 1,
-		},
 
-	}`
-	obj3 := &unstructured.Unstructured{Object: map[string]interface{}{}}
-	if err := yaml.Unmarshal([]byte(y3), &obj3.Object); err != nil {
-		b.Fatalf("Failed to parse yaml object: %v", err)
+	for _, test := range tests {
+		b.Run(test.gvk.Kind, func(b *testing.B) {
+			f := NewTestFieldManager(test.gvk)
```

Comment:
Is it okay to create this here and update it within each of the b.Run cases below?

Reply:
Yeah, I think it is. It's also using `f.Reset` so that it doesn't save the "live object".

Reply:
Sure, I was just worried if this would be run in parallel